homeassistant/components/motionblinds_ble/__init__.py

```diff
@@ -0,0 +1,104 @@
+"""Motionblinds BLE integration."""
+
+from __future__ import annotations
+
+from functools import partial
+import logging
+
+from motionblindsble.const import MotionBlindType
+from motionblindsble.crypt import MotionCrypt
+from motionblindsble.device import MotionDevice
+
+from homeassistant.components.bluetooth import (
+    BluetoothCallbackMatcher,
+    BluetoothChange,
+    BluetoothScanningMode,
+    BluetoothServiceInfoBleak,
+    async_ble_device_from_address,
+    async_register_callback,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_ADDRESS, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.event import async_call_later
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_BLIND_TYPE, CONF_MAC_CODE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [
+    Platform.COVER,
+]
+
+CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Motionblinds BLE integration."""
+
+    _LOGGER.debug("Setting up Motionblinds BLE integration")
+
+    # The correct time is needed for encryption
+    _LOGGER.debug("Setting timezone for encryption: %s", hass.config.time_zone)
+    MotionCrypt.set_timezone(hass.config.time_zone)
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Motionblinds BLE device from a config entry."""
+
+    _LOGGER.debug("(%s) Setting up device", entry.data[CONF_MAC_CODE])
+
+    hass.data.setdefault(DOMAIN, {})
+
+    ble_device = async_ble_device_from_address(hass, entry.data[CONF_ADDRESS])
+    device = MotionDevice(
+        ble_device if ble_device is not None else entry.data[CONF_ADDRESS],
+        blind_type=MotionBlindType(entry.data[CONF_BLIND_TYPE]),
+    )
+
+    # Register Home Assistant functions to use in the library
+    device.set_ha_create_task(
+        partial(
+            entry.async_create_task,
```

Comment:
without knowing what this function does `entry.async_create_background_task` is probably better as this is correctly closed on shutdown. 
A function that is called `set_ha_create_task` is ringing a warning bell though, as the libraries normally don't need to serve anything explicitly for HA.

Reply:
I was told that it was best to use HA wrappers for `asyncio` library functions, therefore I pass HA's `entry.async_create_task` and `async_call_later` such that the motionblinds_ble library uses those functions

Reply:
Or should I use standard `asyncio.create_task` and `asyncio.call_later`?

Reply:
no, it's correct to use the wrappers. I don't know your library, so I don't know your design, but maybe it could make sense to expose the task function and just start it from here? Just something for you to think about... 

`entry.async_create_background_task` is probably still better (compared to `entry.async_create_task`) as it's automatically cancelled when your config entry unloads.

Reply:
Sure, I will have a look at passing `entry.async_create_background_task` instead!

Reply:
For my BLE library I have completely moved all the background tasks regarding connection and status updates into the Home Assistant integration to circumvent this issue. I don't really see a problem with passing the wrappers into the library if this is preferred though.